packages/nvidia_nat_rag_lib/tests/test_rag_lib_function.py

```
@@ -0,0 +1,674 @@
+        from nvidia_rag import NvidiaRAG
+
+        rag = NvidiaRAG()
+        assert rag is not None
+        assert isinstance(rag, NvidiaRAG)
+
+    def test_generate_method_exists(self) -> None:
+        """NvidiaRAG should have a generate method."""
+        from nvidia_rag import NvidiaRAG
+
+        assert hasattr(NvidiaRAG, "generate")
+        assert callable(getattr(NvidiaRAG, "generate"))
```

Ruff (0.14.13)

Do not call `getattr` with a constant attribute value. It is not any safer than normal property access.

Replace `getattr` with attribute access

(B009)